app/models/shipit/commit_deployment.rb

```diff
@@ -0,0 +1,58 @@
+module Shipit
+  class CommitDeployment < ActiveRecord::Base
+    GITHUB_STATUSES = {
+      'pending' => 'pending',
+      'failed' => 'failed',
+      'success' => 'success',
+      'error' => 'error',
+      'aborted' => 'error',
+    }
+
+    belongs_to :commit
+    belongs_to :task
+    has_many :statuses, class_name: 'CommitDeploymentStatus'
+
+    after_commit :schedule_create_on_github, on: :create
+
+    delegate :stack, to: :task
+
+    class << self
+      def append_status(task_status)
+        if github_status = GITHUB_STATUSES[task_status]
+          all.each do |deployment|
```

Comment:
Took me a while to understand that this `all` would be scoped by the task id. Maybe we should add this method as an association extension to make clear that we expect it to be used only in the `Deploy` model?